### PR TITLE
EvForm tracebacks with dissimilar line length

### PR DESCRIPTION
**Pull request by Henddher**
#### Brief overview of PR changes/additions

Force evform.raw_form to have all lines of the same length - effectively a rectangle

#### Other info (issues closed, discussion etc)

See #1459

- [x] bug fixed
- [x] documentation updated

### CODE REVIEW

File: evennia/utils/evform.py [hunk 1]
```diff
@@ -153,6 +153,31 @@
...
+        nlines (list): list of `ANSIString`s of
+        same length as the longest input line
+
+    """
+    maxl = 0
+    for line in lines:
+        if isinstance(line, (ANSIString, basestring)):
+            maxl = max(len(line), maxl)
+        else:
+            raise ValueError()
```
[the remark this answers is not in the document]
Comment (Griatch) [reply evennia/utils/evform.py]: @Henddher I don't think you need to do an `isinstance` check here. If the incoming data is not a string the system will raise an exception further down instead (Python prefers "leap before you look" after all). So you can write this much more succinctly as: 
```python
maxl = max(len(line) for line in lines)
```

File: evennia/utils/evform.py [hunk 2]
```diff
@@ -184,7 +209,7 @@ def __init__(self, filename=None, cells=None, tables=None, form=None, **kwargs):
             filename (str): Path to template file.
             cells (dict): A dictionary mapping of {id:text}
             tables (dict): A dictionary mapping of {id:EvTable}.
-            form (dict): A dictionary of {"CELLCHAR":char,
+            form (dict): A dictionary of {"FORMCHAR":char,
```
[the remark this answers is not in the document]
Comment (Griatch) [reply evennia/utils/evform.py]: Nice catch!

File: evennia/utils/evform.py [hunk 3]
```diff
@@ -153,6 +153,31 @@
...
+    for line in lines:
+        if isinstance(line, (ANSIString, basestring)):
+            maxl = max(len(line), maxl)
+        else:
+            raise ValueError()
+    nlines = []
+    for line in lines:
+        line += ' ' * (maxl - len(line))
+        nlines.append(line)
+    return nlines
```
Comment (Griatch) [inline evennia/utils/evform.py]: This is fine and quite readable. Just pointing out that this too could be done slightly more succinctly (and more efficiently) using a list comprehension: 

```python
return [line + ' ' * (maxl - len(line)) for line in lines]
```

File: evennia/utils/evform.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Henddher) [reply evennia/utils/evform.py]: Technically, it was spotted by @RyanStein in his bug report so kudos to him.

File: evennia/utils/evform.py (hunk 3 above)
Comment (Henddher) [reply evennia/utils/evform.py]: II was going with 

> Simple is better than complex.
[PEP-0020](https://www.python.org/dev/peps/pep-0020/)

Comment (Henddher) [reply evennia/utils/evform.py]: The difference is nice though 👍 

```diff
diff --git a/evennia/utils/evform.py b/evennia/utils/evform.py
index 8a9828bc..3742f50d 100644
--- a/evennia/utils/evform.py
+++ b/evennia/utils/evform.py
@@ -161,21 +161,12 @@ def _to_rect(lines):
         lines (list): list of `ANSIString`s
 
     Returns:
-        nlines (list): list of `ANSIString`s of
+        (list): list of `ANSIString`s of
         same length as the longest input line
 
     """
-    maxl = 0
-    for line in lines:
-        if isinstance(line, (ANSIString, basestring)):
-            maxl = max(len(line), maxl)
-        else:
-            raise ValueError()
-    nlines = []
-    for line in lines:
-        line += ' ' * (maxl - len(line))
-        nlines.append(line)
-    return nlines
+    maxl = max(len(line) for line in lines)
+    return [line + ' ' * (maxl - len(line)) for line in lines]

```

**Comment by Henddher** [comment]
BTW, @Griatch, this PR is ready again :)

**Comment by Griatch** [comment]
@Henddher Ah, thanks - github does not notify that well when a PR is updated after the first review.

**Pull request merged**